### ISSUE TITLE
[Feature request]: Add support for eufy Smart Lock C33

### ISSUE THREAD

**Issue by osoft** [bropat/eufy-security-client#526]
### Describe the solution you'd like

C33 is a new smart lock from Eufy that is designed to be a lever lock.
https://www.eufy.com/products/t85l0111

**Comment by air720boarder** [comment]
+1 for this request as well

**Comment by grimgrinninjay** [comment]
One more request for this.

**Comment by jhongturney** [comment]
https://github.com/bropat/eufy-security-client/pull/528

I just need to determine the actual value for the entry in `types.ts`

**Comment by llukezz** [comment]
+1 for this device! Looks like you are already working on it, thank you!

**Comment by bropat** [comment]
I need someone to share the device with me.

**Comment by osoft** [comment]
> I need someone to share the device with me.

Let me know how to

**Comment by jhongturney** [comment]
@bropat I just updated my PR with the actual ID number, got it from the dev console 👍🏻

**Comment by bropat** [comment]
@jhongturney 

Can you share your device with me so that I can test everything and, if necessary, implement the missing part straight away?

**Comment by jhongturney** [comment]
> [@jhongturney](https://github.com/jhongturney)
> 
> Can you share your device with me so that I can test everything and, if necessary, implement the missing part straight away?

Yes, I can do that, can you let me know where to send the invite?

**Comment by bropat** [comment]
patrick.broetto+us2@gmail.com

**Comment by jhongturney** [comment]
@bropat sent, let me know if you need anything else 👍🏻

**Comment by bropat** [comment]
@jhongturney 

I need admin rights.

**Comment by jhongturney** [comment]
@bropat sorry about that, done now!

**Comment by bropat** [comment]
@jhongturney 

Thank you!

Unfortunately, I already have some bad news... This device does not use P2P communication at all (is the first one I see :() for controlling the device.
All communication is via MQTTS (security-mqtt-us.anker.com).
So the device is not compatible with the current implementation.

**Comment by bropat** [comment]
The only thing that can be done at the moment is to poll the device data via the Eufy Cloud without being able to make any changes.

**Comment by jhongturney** [comment]
Interesting, and I did think that I saw it was MQTTS, but I was optimistic 😅 

If I have time I'll look into this further.

**Comment by sriegodedios** [comment]
One more for this request

**Comment by gbrogna** [comment]
Hello.  I need this integration.  Is there anything I can do to help?

**Comment by InfoSecured** [comment]
+1 here as well.

**Comment by jmferris** [comment]
I'm hoping that this is provided via a firmware update, as well.

Not sure if it will help, but it certainly won't hurt - I've submitted a feedback request to Eufy, explaining that we would like to see P2P support added.  It only takes a minute, and the more people who ask, I would hope the better the chances that they would consider it.

**Comment by weberjam007** [comment]
> I'm hoping that this is provided via a firmware update, as well.
> 
> Not sure if it will help, but it certainly won't hurt - I've submitted a feedback request to Eufy, explaining that we would like to see P2P support added. It only takes a minute, and the more people who ask, I would hope the better the chances that they would consider it.

Did you just email support@eufy.com or did you find a feedback form? I want to send my feedback and ensure it follows the same communication channel as yours.

**Comment by jmferris** [comment]
> Did you just email [support@eufy.com](mailto:support@eufy.com) or did you find a feedback form? I want to send my feedback and ensure it follows the same communication channel as yours.

I meant to mention that.  Thanks for catching it!  I used the form on their website, using the "Feedback" option.  It can be found here:

[EmailContactUs | eufy support](https://support.eufy.com/s/emailcontactus)

**Comment by swooshant** [comment]
T85L0 / C33 lock!
+1

**Comment by nick-pape** [comment]
@bropat After working extensively with Claude over the last couple days I was finally able to reverse engineer the secure MQTT approach that's being used with all these newer locks! Really, really exciting. It's definitely super indirect. Got it working with a script, and working to integrate that into my fork: https://github.com/nick-pape/eufy-security-client . Once I get some time to review this code I'll post a PR and a spec back to eufy-security-client.

**Comment by max246** [comment]
> So, I have the T85D0 lock, here's what I added on my branch to get this working in HA: [nick-pape#2](https://github.com/nick-pape/eufy-security-client/pull/2)
> 
> And the detailed description for how the MQTT works: https://github.com/nick-pape/eufy-security-client/blob/6df1087c55b2a41c156d9bc6f46155897050bf91/docs/t85d0-smart-lock-protocol.md
> 
> [@bropat](https://github.com/bropat) . And yeah, since this is AI generated code, I'm going to do some review and cleanup before creating a PR back here.

Merged something in based on a old PR that was never updated. please have a look whatever is missing from yours an open one :)

**Comment by nick-pape** [comment]
@max246 sorry, to be clear, did you merge an sMQTT/auth flow or something else? Want to know what to look for to update #797

**Comment by max246** [comment]
I didn't merge a sMqtt , there was a pr regarding the lock and just needed a rebase but the contributor didn't reply for weeks so I just went ahead and rebase and merged. 

Might work to enable some functionality but might missing others due to the need of sMqtt.

I am just tidy up all the open pr since some are ready but needs more attention

**Comment by max246** [comment]
Fixed 3.8.0
